.changeset/giant-bees-speak.md

```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-techdocs-backend': minor
+---
+
+**BREAKING**: The `cache` option is now required by `createRouter`.
+
+Added catalog-based authorization to TechDocs backend. When permissions are enabled for Backstage (via the `permission.enabled` config) the current user must have read access to the doc's corresponding catalog entity. The backend will return a 404 if the current user doesn't have access.
```

Comment:
it might be helpful to connect this change back to why the cache is now required, it seems like a bit random otherwise if only looking at the changeset.

maybe even just more or less copy pasting the comment you have below:
```
// Entities are cached to optimize the /static/docs request path, which can be called many times
// when loading a single techdocs page.
```